### PR TITLE
load last 3 months of sermons, then load the rest in the background. …

### PR DESCRIPTION
**Pull request by loukkevin**
…Added loading indicator

### CODE REVIEW

**Comment by loukkevin** [comment]
Not sure what the package-lock.json is all about, maybe it's part of create-react-app. Hopefully it doesn't break the prod build somehow.

**Comment by smcadams86** [comment]
> 
> 
> Not sure what the package-lock.json is all about, maybe it's part of create-react-app. Hopefully it doesn't break the prod build somehow.

package lock is a way of pinning down the node dependencies that you are using so you don't inadvertently upgrade a package when you don't intend to. it should be fine.

**Comment by smcadams86** [comment]
hmm. so I fired this up locally. I don't see a loading indicator when either on initial page load or while scrolling.
It also doesn't seem like the loading is delegated to a background task - the browser still hangs waiting for data to load as we are not loading in a proper async manner.

**Comment by loukkevin** [comment]
Yeah, it's not technically in the background. It is a bit janky, but for me it loaded initial sermons in under a second, they were playable, and then loaded the rest after awhile. Interesting that you're not seeing the indicator. Maybe it just loads too fast?

**Comment by smcadams86** [comment]
No it loads very slowly. 10 seconds+

File: src/App.js [hunk 1]
```diff
@@ -9,53 +10,78 @@ var base = Rebase.createClass({
...
       asArray: true
     });
 
-    this.sermonsRef = base.bindToState('sermons', {
+    //initially load last 3 months of sermons
+    var currentTime = new Date();
+    var year = currentTime.getFullYear();
+    var month = currentTime.getMonth();
+    var day = currentTime.getDay();
+    var prev3Months = month > 3 ? 1 : month - 3;
```
Comment (smcadams86) [inline src/App.js:38]:
```suggestion
    var prev3Months = month < 4 ? 1 : month - 3;
```

Comment (smcadams86) [reply src/App.js:38]: image

Comment (loukkevin) [reply src/App.js:38]: It doesnt work that way.

File: src/App.js [hunk 2]
```diff
@@ -9,53 +10,78 @@ var base = Rebase.createClass({
...
+
+    var endDateString = new Date(year, month + 1, day).toISOString();
+    var startDateString = new Date(year, prev3Months+1, 1).toISOString();
```
Comment (smcadams86) [inline src/App.js:41]:
```suggestion

    //initially load last 3 months of sermons
    var endDateString = moment().format();
    var startDateString = moment().subtract(3, 'months').format();
```

File: src/App.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
-import React, { Component } from 'react';
-import Rebase from 're-base';
+import React, { Component } from "react";
+import Rebase from "re-base";
+import LoadingIndicator from "./modules/common/LoadingIndicator";
```
Comment (smcadams86) [inline src/App.js:4]:
```suggestion
import moment from "moment";
```

File: src/App.js [hunk 4]
```diff
@@ -9,53 +10,78 @@ var base = Rebase.createClass({
...
 
+  componentDidMount() {
+    //get the rest of the data
+    var currentTime = new Date();
+    var year = currentTime.getFullYear();
+    var month = currentTime.getMonth();
+    var day = currentTime.getDay();
+
+    var endDateString = new Date(year, month + 1, day).toISOString();
+    var startDateString = new Date(2012, 0, 1).toISOString();
```
Comment (smcadams86) [inline src/App.js:71]:
```suggestion
    //get the rest of the data
    var endDateString = moment().format();
    var startDateString = moment().subtract(10, 'years').format();

```

**Pull request merged**